### PR TITLE
Fix to use Dagger 2 Inject pattern.

### PR DESCRIPTION
**Pull request by rogerhu**
Flatten to make more clear.

### CODE REVIEW

File: app/src/main/java/com/codepath/daggerexample/MainActivity.java [hunk 1]
```diff
@@ -25,8 +31,8 @@
     @Inject
     Retrofit mRetrofit;
 
-    private GitHubApi mGitHubApi;
```
Comment (rogerhu) [inline app/src/main/java/com/codepath/daggerexample/MainActivity.java:28]: This was basically bypassing everything that Dagger 2 is good for.

**Comment by nesquena** [comment]
Thanks

File: app/src/main/java/com/codepath/daggerexample/MainActivity.java (hunk 1 above)
Comment (Isanderthul) [inline app/src/main/java/com/codepath/daggerexample/MainActivity.java:28]: the docs and the code are not in alignment... https://github.com/codepath/android_guides/wiki/Dependency-Injection-with-Dagger-2, should this be made an issue?

File: app/src/main/java/com/codepath/daggerexample/MainActivity.java [hunk 2]
```diff
@@ -39,24 +45,31 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                        public void onFailure(Throwable t) {
+
+                        }
+                    });
+                }
+
+            });
 
         ((MyApp) getApplication()).getGitHubComponent().inject(this);
-        mGitHubApi = new GitHubApi(mRetrofit);
```
Comment (rogerhu) [inline app/src/main/java/com/codepath/daggerexample/MainActivity.java:59]: The issue is here.  You are avoiding dependency injection and inserting Retrofit into here when it should be a provider call:

https://github.com/codepath/android_guides/wiki/Dependency-Injection-with-Dagger-2#component-dependencies

``` java
@Module
public class GitHubModule {

    public interface GitHubApiInterface {
      @GET("/org/{orgName}/repos")
      Call<ArrayList<Repository>> getRepository(@Path("orgName") String orgName);
    }

    @Provides
    @UserScope // needs to be consistent with the component scope
    public GitHubApiInterface providesGitHubInterface(Retrofit retrofit) {
        return retrofit.create(GitHubApiInterface.class);
    }
}

```

**Pull request merged**